masq_lib/src/percentage.rs

```diff
@@ -227,91 +141,271 @@ impl PurePercentage {
-    fn handle_upper_overflow<N>(&self, num: N) -> N
+    pub fn increase_by_percent_for<N>(&self, num: N) -> N
     where
         N: PercentageInteger,
         <N as TryFrom<i8>>::Error: Debug,
         i16: TryFrom<N>,
         <i16 as TryFrom<N>>::Error: Debug,
     {
-        let hundred = N::try_from(100).expect("Each type has 100");
-        let modulo = num % hundred;
-        let percent = N::try_from(self.degree as i8).expect("Each type has 100");
-
-        let without_treated_remainder = (num / hundred) * percent;
-        let final_remainder_treatment = Self::treat_remainder(modulo, percent);
-        without_treated_remainder + final_remainder_treatment
+        self._increase_by_percent_for(num)
     }
 
-    fn treat_remainder<N>(modulo: N, percent: N) -> N
+    pub fn decrease_by_percent_for<N>(&self, num: N) -> N
     where
         N: PercentageInteger,
         <N as TryFrom<i8>>::Error: Debug,
         i16: TryFrom<N>,
         <i16 as TryFrom<N>>::Error: Debug,
     {
-        let extended_remainder_prepared_for_rounding = i16::try_from(modulo)
-            .unwrap_or_else(|_| panic!("u16 from -100..=100 failed at modulo {:?}", modulo))
-            * i16::try_from(percent).expect("i16 from within 0..=100 failed at multiplier");
-        let rounded = Self::div_by_100_and_round(extended_remainder_prepared_for_rounding);
-        N::try_from(rounded as i8).expect("Each type has 0 up to 100")
+        self._decrease_by_percent_for(num)
+    }
+}
+
+fn base_and_rem_for_ensured_i16<N>(a: N, b: N) -> (N, N)
+where
+    N: PercentageInteger,
+    <N as TryFrom<i8>>::Error: Debug,
+    i16: TryFrom<N>,
+    <i16 as TryFrom<N>>::Error: Debug,
+{
+    let num = i16::try_from(a)
+        .expect("Remainder: Each integer can go up to 100, or down to -100 if signed")
+        * i16::try_from(b)
+            .expect("Percents: Each integer can go up to 100, or down to -100 if signed");
+
+    let (base, remainder) = base_and_rem_from_div_100(num);
+
+    (
+        N::try_from(base as i8)
+            .expect("Base: Each integer can go up to 100, or down to -100 if signed"),
+        N::try_from(remainder as i8)
+            .expect("Remainder: Each integer can go up to 100, or down to -100 if signed"),
+    )
+}
+
+fn base_and_rem_from_div_100<N>(num: N) -> (N, N)
+where
+    N: PercentageInteger,
+    <N as TryFrom<i8>>::Error: Debug,
+{
+    let hundred = N::try_from(100i8).expect("Each integer has 100");
+    let modulo = num % hundred;
+    (num / hundred, modulo)
+}
+
+// This is a wider type that allows to specify cumulative percents of more than only 100.
+// The expected use of this would look like requesting percents meaning possibly multiples of 100%,
+// roughly, of a certain base number. Similarly to the PurePercentage type, also signed numbers
+// would be accepted.
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct LoosePercentage {
+    multiplier_of_100_percent: u32,
+    degrees_from_remainder: PurePercentage,
```

Comment:
Instead of calling `multiplier_of_100_percent` you can simply call it `quotient` and `degrees_from_remainder` to `remainder`.